spec/fastsync/vdd/fastsyncL2.tla

```diff
@@ -0,0 +1,679 @@
+    (* Corresponds to function prunablePeers the in scheduler.go file. Note that this function only checks if block has been  *)
+    (* received from a peer during peerTimeout period.                                                                        *)
+    (* Note that in case no request has been scheduled to a correct peer, or a request has been scheduled                     *)
+    (* recently, so the peer hasn't responded yet, a peer will be removed as no block is received within peerTimeout.         *)
+    (* In case of faulty peers, we don't have any guarantee that they will respond.                                           *)
+    (* Therefore, we model this with nondeterministic behavior as it could lead to peer removal, for both correct and faulty. *)
+    (* See scheduler.go                                                                                                       *)
+    (* https://github.com/tendermint/tendermint/blob/4298bbcc4e25be78e3c4f21979d6aa01aede6e87/blockchain/v2/scheduler.go#L335 *)
+    LET toRemovePeers == IF nonDetFlag THEN bPool.peerIds ELSE {} IN
+       
+    (*
+      Corresponds to logic for pruning a peer that is responsible for delivering block for the next height.
+      If a peer responsible for next height is correct, we don't remove it as logic is based on difference between pending time
+      and current time. In the synchronous system model, correct peer will always respond timely. 
+      See scheduler.go     
+      https://github.com/tendermint/tendermint/blob/4298bbcc4e25be78e3c4f21979d6aa01aede6e87/blockchain/v2/scheduler.go#L617    
+    *)
+    LET nextHeightPeer == bPool.pendingBlocks[bPool.height] IN
+    LET prunablePeers == 
+        IF /\ nextHeightPeer /= NilPeer
+           /\ nextHeightPeer \in FAULTY
+           /\ nonDetFlag 
+        THEN toRemovePeers \union {nextHeightPeer} 
+        ELSE toRemovePeers
+    IN
+    RemovePeers(prunablePeers, bPool)
+    
+
+\* Handle SyncTimeout. It models if progress has been made (height has increased) since the last SyncTimeout event.
+HandleSyncTimeout(bPool) ==    
+    [bPool EXCEPT 
+            !.syncedBlocks = bPool.height - bPool.syncHeight, 
+            !.syncHeight = bPool.height
+    ]      
+
+
+\* Handle control messages.
+HandleControlMsg(msg, bPool) == 
+    IF msg.type = "statusResponse" 
+    THEN HandleStatusResponse(msg, bPool)
+    ELSE IF msg.type = "addPeer" 
+         THEN AddPeer(msg.peerId, bPool)
+         ELSE IF msg.type = "removePeer"
+         THEN RemovePeer(msg.peerId, bPool)  
+         ELSE IF msg.type = "syncTimeout"
+              THEN HandleSyncTimeout(bPool)  
+              ELSE bPool 
+
+HandleResponse(msg, bPool) == 
+    IF msg = NoMsg 
+    THEN bPool
+    ELSE IF msg.type = "blockResponse" 
+         THEN HandleBlockResponse(msg, bPool) 
+         ELSE HandleControlMsg(msg, bPool)  
+    
```

Comment:
The two functions above could be merged and maybe aligned better, e.g.
```
HandleResponse(msg, bPool) == 
    IF msg.type = "blockResponse" THEN 
      HandleBlockResponse(msg, bPool) 
    ELSE IF msg.type = "statusResponse" THEN 
      HandleStatusResponse(msg, bPool) 
    ELSE IF msg.type = "addPeer" THEN 
      AddPeer(msg.peerId, bPool) 
    ELSE IF msg.type = "removePeer" THEN 
      RemovePeer(msg.peerId, bPool)  
    ELSE IF msg.type = "syncTimeout" THEN 
      HandleSyncTimeout(bPool)
    ELSE 
      bPool
```

or use boolean expressions.